java/flight/src/main/java/org/apache/arrow/flight/FlightDescriptor.java

```diff
@@ -135,7 +159,7 @@ public boolean equals(Object obj) {
       if (other.cmd != null) {
         return false;
       }
-    } else if (!cmd.equals(other.cmd)) {
+    } else if (!Arrays.equals(cmd, other.cmd)) {
```

Comment:
Objects.deepEquals()?

Reply:
`cmd` is a byte array, so I don't see a reason to prefer the Object-typed method, unless it's stylistic or idiomatic to use it? (I'm not too familiar with Java intricacies)

Reply:
For some reason I thought cmd was more complex. Nevermind :)